src/libraries/System.Private.CoreLib/src/System/Array.cs

```diff
@@ -1115,31 +1116,31 @@ public static int IndexOf<T>(T[] array, T value, int startIndex, int count)
                 if (Unsafe.SizeOf<T>() == sizeof(byte))
                 {
                     int result = SpanHelpers.IndexOf(
-                        ref Unsafe.Add(ref array.GetRawSzArrayData(), startIndex),
+                        ref Unsafe.Add(ref MemoryMarshal.GetRawArrayData(Unsafe.As<byte[]>(array)), startIndex),
```

Comment:
Can you explain why this change is better? Is the goal to avoid using `GetRawSzArrayData`?

Reply:
This PR removes the internal `GetRawSzArrayData` method in favor of the new public `MemoryMarshal` API. This particular call site has a bit of a smell unfortunately since the input isn't already of type `T[]`, but the majority of the call sites throughout the PR do look cleaner. See below for a canonical example of how most call sites no longer need `Unsafe.As` with this new API.

https://github.com/dotnet/runtime/pull/1036/files#diff-d1745c212e8b7f4a46bc702c8d511cadR41